ServerlessLibraryAPI/ClientApp/src/components/DetailView/ActionBar.js

```diff
@@ -1,19 +1,61 @@
 import React, { Component } from "react";
 import { Icon, Link as FabricLink } from "office-ui-fabric-react";
+import { trackEvent } from "../../helpers";
 import "./ActionBar.scss";
 
 class ActionBar extends Component {
+  constructor(props) {
+    super(props);
+    this.outboundRepoClick = this.outboundRepoClick.bind(this);
+    this.outboundDeployClick = this.outboundDeployClick.bind(this);
+    this.openInVSCodeClick = this.openInVSCodeClick.bind(this);
+  }
+
+  outboundDeployClick(template) {
+    let eventData = this.getDeployClickEventDataToLog(template);
+    trackEvent("/sample/deploy/agree", eventData);
+  }
+
+  getDeployClickEventDataToLog(template) {
+    return {
+      template: template
```

Comment:
please also include the id in the event data

Reply:
in all events? If emit Id, no need to emit enything else. What is ur view?